packages/indy-vdr/src/pool/IndyVdrPool.ts

```diff
@@ -0,0 +1,182 @@
+      data: TypedArrayEncoder.fromString(request.signatureInput),
+      key: signingKey,
+    })
+
+    request.setSignature({
+      signature,
+    })
+
+    return await this.pool.submitRequest(request)
+  }
+
+  public async submitReadRequest<Request extends IndyVdrRequest>(request: Request) {
+    return await this.pool.submitRequest(request)
+  }
+
+  private async appendTaa(request: IndyVdrRequest) {
+    const authorAgreement = await this.getTransactionAuthorAgreement()
+    const poolTaa = this.config.transactionAuthorAgreement
+
+    // If ledger does not have TAA, we can just send request
+    if (authorAgreement == null) {
+      return request
+    }
+
+    // Ledger has taa but user has not specified which one to use
+    if (!poolTaa) {
+      throw new IndyVdrError(
+        `Please, specify a transaction author agreement with version and acceptance mechanism. ${JSON.stringify(
+          authorAgreement
+        )}`
+      )
+    }
+
+    // Throw an error if the pool doesn't have the specified version and acceptance mechanism
+    if (
+      authorAgreement.version !== poolTaa.version ||
+      !authorAgreement.acceptanceMechanisms.aml[poolTaa.acceptanceMechanism]
+    ) {
+      // Throw an error with a helpful message
+      const errMessage = `Unable to satisfy matching TAA with mechanism ${JSON.stringify(
+        poolTaa.acceptanceMechanism
+      )} and version ${poolTaa.version} in pool.\n Found ${JSON.stringify(
+        authorAgreement.acceptanceMechanisms.aml
+      )} and version ${authorAgreement.version} in pool.`
+      throw new IndyVdrError(errMessage)
+    }
+
+    const acceptance = indyVdr.prepareTxnAuthorAgreementAcceptance({
+      text: authorAgreement.text,
+      version: authorAgreement.version,
+      taaDigest: authorAgreement.digest,
+      time: Math.floor(new Date().getTime() / 1000),
+      acceptanceMechanismType: poolTaa.acceptanceMechanism,
+    })
+
+    request.setTransactionAuthorAgreementAcceptance({ acceptance })
+  }
+
+  private async getTransactionAuthorAgreement(): Promise<AuthorAgreement | null> {
+    // TODO Replace this condition with memoization
+    if (this.authorAgreement !== undefined) {
+      return this.authorAgreement
+    }
+
+    const taaRequest = new GetTransactionAuthorAgreementRequest({})
+    const taaResponse = await this.submitReadRequest(taaRequest)
+
+    const acceptanceMechanismRequest = new GetAcceptanceMechanismsRequest({})
+    const acceptanceMechanismResponse = await this.submitReadRequest(acceptanceMechanismRequest)
+
+    const taaData = taaResponse.result.data
+
+    // TAA can be null
+    if (taaData == null) {
+      this.authorAgreement = null
+      return null
+    }
+
+    // If TAA is not null, we can be sure AcceptanceMechanisms is also not null
+    const authorAgreement = taaData as Omit<AuthorAgreement, 'acceptanceMechanisms'>
+    const acceptanceMechanisms = acceptanceMechanismResponse.result.data as AcceptanceMechanisms
```

Comment:
Cast to unknown first until we have a new release  for indy-vdr

```suggestion
     // FIME: remove cast when https://github.com/hyperledger/indy-vdr/pull/142 is released
    const acceptanceMechanisms = acceptanceMechanismResponse.result.data as unknown as AcceptanceMechanisms
```